### PR TITLE
PRNGKeyArrayImpl: add aval property

### PR DESCRIPTION
**Pull request by jakevdp**
This makes it more readily compatible with jax.numpy routines.

### CODE REVIEW

File: jax/_src/lax/control_flow/loops.py [hunk 1]
```diff
@@ -22,7 +22,7 @@
 import weakref
 from jax._src import core
 from jax._src import linear_util as lu
-from jax import config
+from jax import config  # type: ignore[no-redef]
```
Comment (jakevdp) [inline jax/_src/lax/control_flow/loops.py:25]: This is unrelated, but for some reason this causes a false-positive in mypy.

**Review by froystig — APPROVED** [review]
Test?

**Comment by jakevdp** [comment]
I think since `pytype_aval_mappings` uses the attribute now, existing test coverage is sufficient, what do you think?

**Comment by froystig** [comment]
Sounds good, merge away!

**Pull request closed without merging**